### ISSUE TITLE
FITCount App Beta/Live bundle

### ISSUE THREAD

**Issue by JimChiazzese1** [NERC-CEH/fit-count-app#157]
As per email, bundling live/beta updates and required changes.

That aside, to summarise the next bundle of changes…

- [x]	Chile and Argentina can go to Beta
- [x]	Croatia has some picture credits to add and then can also go to Beta (see below)
- [ ]	Germany translations are fixed and go live
- [ ]	Portugal still working on their final images so no current action

Croatia image credits to add:

Igor Boršić
Tamara Kirin
Ana Ješovnik 
Fir0002 via Wikimedia

**Comment by kazlauskis** [comment]
We've added Croatia to Beta now. It will be ready sometime this week when the app stores approve it. To push Germany's update, we have to do the same as before: take the Argentina + Chile + Croatia Beta release down and release Germany to live and re-release the Beta with Croatia and others. Would you like us to do that?

I was looking for a better way to manage the releases. Most projects don't mind if a new country is released without much Beta testing, so this is the first time we have had this problem. We can change the setup if we want a long-running Beta track while simultaneously releasing hotfixes and content updates to live. Unfortunately, there isn't an easy way to detect if an app is running in Beta vs live, but we can use an external service that would allow us to manage these flows. So, enable some features (like new countries) for specific users only. We could build this service or use some existing one.

**Comment by JimChiazzese1** [comment]
Thanks Karolis – Apologies for the silence, I picked up another infection of some sort (life with kids!) and ended up in hospital again for a few days and then home, but out of action for a week.
Finally getting back to normal and picking things up again.

I saw Croatia on the Beta on my phone, so that’s great.

Germany definitely needs to be live as soon as possible and Croatia and Chile should be at Beta – They’ve both been informed that should just be for a very limited time and they’re happy with that.
In which case, before I let Croatia know Beta is available, do you want to push Germany live (as I assume this means Croatia coming off Beta for a short time)?

Thanks

Jim

**Comment by kazlauskis** [comment]
@JimChiazzese1, sorry to hear you've been off sick, I can fix the German texts and push it to live as you say, but we should review how we make these updates.

It would be better if we didn't push new countries to Beta until all the new country's content is ready and people confirm it is complete (including translations) by looking at the demo web app. Beta testing is helpful mainly for testing new functionality that can break on different devices and should be less about checking the content. So we should remove the Beta testing stage for most of our app updates. At least, this is what we do in most of our other projects regarding content changes.

I suggest that we remove Argentina+Chile+Croatia from Beta and push the new German translation update to live. Once, the new countries are happy with the demo app's content, we'll push it directly to live. The alternative would be to correct the German translations and push everything to live now. I wonder if anyone will find and actively use the app in those new countries without some promotion first.

**Comment by JimChiazzese1** [comment]
I can understand the country's need/want for a Beta even though, as you say, it's unlikely any end users will start using the live app without being directed to it or promotion. 
All the countries (in theory) fully review and confirm they're happy with the online demo before the next stage, but the beta gives them the opportunity to test it in action. 
In theory though that is unnecessary, we know the app works as it's already in wide use, so really all the beta does is let them see the demo in app form, although it does also give the country coordinator chance to test content download etc.
I'm happy to go with whatever you suggest on this. We could go straight from online demo sign off to live if that works better?
Of course the other danger with having a beta stage is that they don't check the online demo properly - like Germany, how did they not spot the translation issue?!

If you think we're better to go direct from online demo to live, and just keep beta for feature devevlopment/testing, then I'm happy to convey that to the countries. As we've said, the chances of people discovering it before it's promoted is minimal and even if they do, that's not really an issue. Although it may only be an issue if we can't tie down the global terms and privacy agreement, but we're nearly there on that.

In short then, I think we can push Croatia, Chile and Germany live. However, Argentina aren't ready for anything currently, so we can remove that from Beta.

Thanks Jim (sorry for the long reply!)

**Comment by JimChiazzese1** [comment]
Sorry, one other thing comes to mind @kazlauskis - there is of course a step between online confirmation and going live, which is getting the new entries (habitats, flowers, insect types) set-up in the warehouse. I'll need to request that for Chile/Croatia before going live.

**Comment by kazlauskis** [comment]
You're right, all the warehouse attributes should be present along with translations or other content before we push an app update Beta/Live. 

I've removed the 3 Beta countries and pushed Germany live now. Their corrected translations will be public after app stores approve the new release in a few days.

Chile, Argentina and Croatia can test the app in the field by opening this demo link in their device browsers: https://fit.app.flumens.io

It requires a network connection but other than that, it will work almost the same as the Beta release. Once a country is happy with its contents, I can then push the country directly to live.

I am closing this ticket. We can create individual country tickets when it's time to release live or change something.
